### DISCUSSION TITLE
Hardware Debugging: Connecting ST Link to serial monitor in VS code.

### DISCUSSION THREAD

**Discussion by danialwaleed** [bitcraze/discussions#1070] [Q&A]
Hi Everyone,

I am planning to run a custom firmware on crazyflie 2.1. I went through all the steps given in (https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/openocd_gdb_debugging/) to get WSL working and using VS Code. I would like to now get all the debug information in VS Code as well. To do so I have activated the option in the crazyflie firmware, but I am having issues connecting the STM link to the serial monitor in VS Code.  In particular, in the serial monitor extension in VS Code I cannot see the STM USB to connect to so that I can  see/dev/ttyS0 and similar things but none of them correspond to the STM link. Does anyone have any suggestions.

**Comment by knmcguire** [comment]
hi! So you tried the whole USBipd route as well? are you able to see the stlink in the ubuntu terminal with lsusb?
